fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBottomSheet.kt

```diff
@@ -13,63 +15,148 @@ import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
 import androidx.compose.material.Divider
 import androidx.compose.material.Icon
 import androidx.compose.material.IconButton
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
 import androidx.compose.ui.semantics.heading
 import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextDecoration
 import androidx.compose.ui.unit.dp
 import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.BetaLabel
+import org.mozilla.fenix.compose.LinkText
+import org.mozilla.fenix.compose.LinkTextState
 import org.mozilla.fenix.compose.annotation.LightDarkPreview
 import org.mozilla.fenix.compose.button.PrimaryButton
-import org.mozilla.fenix.compose.button.SecondaryButton
+import org.mozilla.fenix.compose.button.TextButton
 import org.mozilla.fenix.theme.FirefoxTheme
 
 /**
  * Firefox Translations bottom sheet dialog.
  */
 @Composable
-fun TranslationsDialogBottomSheet(onSettingClicked: () -> Unit) {
+fun TranslationsDialogBottomSheet(
+    learnMoreUrl: String,
+    showFirstTimeTranslation: Boolean,
+    onSettingClicked: () -> Unit,
+    onLearnMoreClicked: () -> Unit,
+    onTranslateButtonClick: () -> Unit,
+) {
+    var orientation by remember { mutableIntStateOf(Configuration.ORIENTATION_PORTRAIT) }
+
+    val configuration = LocalConfiguration.current
+
+    LaunchedEffect(configuration) {
+        snapshotFlow { configuration.orientation }.collect { orientation = it }
+    }
+
     Column(
-        modifier = Modifier
-            .padding(16.dp),
+        modifier = Modifier.padding(16.dp),
     ) {
-        TranslationsDialogHeader(onSettingClicked)
+        BetaLabel(
+            modifier = Modifier
+                .padding(bottom = 8.dp)
+                .clearAndSetSemantics {},
+        )
+
+        TranslationsDialogHeader(onSettingClicked, showFirstTimeTranslation)
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        if (showFirstTimeTranslation) {
+            TranslationsDialogInfoMessage(onLearnMoreClicked, learnMoreUrl)
+        }
 
         Spacer(modifier = Modifier.height(14.dp))
 
-        Column {
+        when (orientation) {
+            Configuration.ORIENTATION_LANDSCAPE -> {
```

Comment:
If it is an easy change here, could you also adjust the span of the popup in landscape too? On the screen on my test device (Pixel 4a), there is about a 15% gap on each side where I can see the content behind it, instead of it covering  [edge-to-edge](https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?type=design&node-id=7039-32721&mode=design&t=doDKIsA4zuDjY0hy-0).

Additionally, could you adjust the height of the pop-up in landscape some too? On my test device, the landscape pop-up starts at the same height as the toolbar instead of showing the full content, so content is hidden a bit.

Otherwise, we can just follow-up in [1868991](https://bugzilla.mozilla.org/show_bug.cgi?id=1868991).

Reply:
I made the changes only for  the height of the pop-up in landscape.

Reply:
Thanks, height looks good! I'll add the span issue as something for us to follow up in [1868991](https://bugzilla.mozilla.org/show_bug.cgi?id=1868991).